doc/source/Code-Style-Guide.rst

```diff
@@ -0,0 +1,746 @@
+                   An example cube for processing.
+
+           Returns:
+               Output cube after multiplying the input cube by the
+               simple variable.
+           """
+           # Inline comments can be added, if required.
+           cube = self.my_static_method(cube)
+           (max_val, min_val) = self.my_static_method_multiple_returns(cube)
+           new_cube = cube
+           if min_val > 0.0 and max_val < 100.0:
+               new_simple_variable = self.my_method()
+               new_cube = cube * new_simple_variable
+           return new_cube
+
+Helper Functions/Methods
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+Helper functions or methods may live in one of several places. Where
+they should live depends on how they will be used.
+
+**Case 1:** A function used in more than one module. - In this case the
+function should be located in a shared location e.g. utilities.py
+
+**Case 2:** A function used by several classes within one module. - The
+function should be kept outside of any one class, but within the module.
+
+**Case 3:** A function used only within a single class. - Should be kept
+within the class; as a static method if it makes no use of self.
+
+Command Line Interface (CLI)
+----------------------------
+
+Add a command line interface (improver/cli/.py) to invoke plugins that
+can be used as a standalone utility or executable within a suite context
+(e.g. wind downscaling, neighbourhood processing, spot data extraction).
+These CLIs are invoked using ``bin/improver <cli-name>`` (note that the
+CLI filename uses underscores, but the call to use the CLI uses hyphens)
+
+IMPROVER CLIs should only have ``from improver import cli`` as the top
+level imports. Other imports are placed inside the function that uses
+them. This gives the benefit of a more rapid response to the command
+``bin/improver <cli-name> -h`` when those other (often slow) imports are
+not needed.
+
+Each CLI should have a process function. This will require a
+``@cli.clizefy`` decorator to gain the functionality of
+`clize <https://clize.readthedocs.io/en/stable/>`__. If you want the CLI
+to save a cube to disk, it will need the decorator ``@cli.with_output``,
+this will mean on the command line, the ``--output`` flag can be used to
+specify an output path.
+
+To load the cubes, each cube argument will need a type. For a basic cube
+this will be ``cube: cli.inputcube``. If there is a default argument to
+go with the typed variable, spaces are required around the ``=`` for
+example ``weights: cli.inputcube = None``. There are other types which
+can be used such as: - the python standards - ``float`` - ``int`` -
+``bool`` - specific additions - ``cli.inputcube`` - Where a string is
+given which is a path to a cube to load - ``cli.inputjson`` - Where a
+string is given which is a path to a json file to load -
+``cli.comma_separated_list`` - This will convert the argument into that
+format and deal with error handling if no conversion is possible.
```

Comment:
Another bulleted list that has been inlined.

Reply:
Fixed.